pyomo/gdp/plugins/bigm.py

```diff
@@ -1004,3 +1004,33 @@ def get_M_value(self, constraint):
         # fails... (That is, it's a bug in the mapping.)
         lower, upper = transBlock.bigm_src[constraint]
         return (lower[0], upper[0])
+
+    def get_all_M_values_by_constraint(self, model):
+        """Returns a dictionary mapping each constraint to a tuple:
+        (lower_M_value, upper_M_value), where either can be None if the
+        constraint does not have a lower or upper bound (respectively).
+
+        Parameters
+        ----------
+        model: A GDP model that has been transformed with BigM
+        """
+        m_values = {}
+        for disj in model.component_data_objects(
+                Disjunct,
+                active=None,
+                descend_into=(Block, Disjunct)):
+            if disj.transformation_block is not None:
+                transBlock = disj.transformation_block()
+                for cons, (lower, upper) in transBlock.bigm_src.items():
+                    m_values[cons] = self.get_M_value(cons)
```

Comment:
This assumes that all Disjuncts were transformed by BigM.  I think you need a check to make sure bigm_src is there...

Reply:
Good point.

Reply:
Oh, wait, just kidding, if it's not transformed then `disj.transformation_block` will be `None`. If it points to something, then the disjunct is transformed, and it's actually a bug in the transformation if it doesn't have the `bigm_src` dictionary.

Reply:
So hull adds `bigm_src`?